package/MDAnalysis/analysis/backends.py

```diff
@@ -0,0 +1,279 @@
+"""Analysis backends --- :mod:`MDAnalysis.analysis.backends`
+============================================================
+
+Provides :class:`BackendBase` base class to implement custom
+backends for :meth:`MDAnalysis.analysis.base.AnalysisBase.run()` and its
+subclasses. Two built-in backend classes classes are provided:
+:class:`BackendMultiprocessing` that supports parallelization via standard
+python ``multiprocessing`` module, and :class:`BackendDask`, that uses the same
+process-based parallelization as :class:`BackendMultiprocessing`, but different
+serialization algorithm (see https://distributed.dask.org/en/latest/serialization.html
+for explanation of ``dask`` serialization algorithms; ``multiprocessing`` uses
+default ``pickle`` serialization).
+"""
+import warnings
+from typing import Callable
+from MDAnalysis.lib.util import is_installed
+
+
+class BackendBase:
+    """Base class for backend implementation. Initializes an instance and performs
+    checks for its validity, such as n_workers and possibly other ones.
+
+    Parameters
+    ----------
+    n_workers : int
+        number of workers (usually, processes) over which the work is split
+
+    Examples
```

Comment:
This example raises an error: `ValueError: n_workers specified twice`. `backend = BackendMultiprocessing(n_workers=2)` also raises the same error when used as a backend here.

Reply:
Also add a warning **ThreadsBackend leads to erroneous results**.

Reply:
Added a warning to the documentation, and refactored the n_workers instantiation a bit, to grab it from backend argument first.